src/gui/src/dbDescriptors.cpp

```diff
@@ -527,6 +529,249 @@ bool DbNetDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+      }
+    }
+
+    for (const auto& [sink_rect, sink_layer] : sinks) {
+      if (sink_rect.intersects(node_pt) && sink_layer == node_layer) {
+        sink_nodes.insert(node);
+      }
+    }
+  }
+
+  // build connectivity map from the wire graph
+  NodeMap node_map;
+  buildNodeMap(&graph, node_map);
+
+  Painter::Color highlight_color = painter.getPenColor();
+  highlight_color.a = 255;
+
+  painter.savePenAndBrush();
+  painter.setPen(highlight_color, true, 4);
+
+  for (const auto* source_node : source_nodes) {
+    for (const auto* sink_node : sink_nodes) {
+      // find the shortest path from source to sink
+      std::vector<odb::Point> path;
+      findPath(node_map, source_node, sink_node, path);
+
+      if (!path.empty()) {
+        odb::Point prev_pt = path[0];
+        for (const auto& pt : path) {
+          if (pt == prev_pt) {
+            continue;
+          }
+
+          painter.drawLine(prev_pt, pt);
+          prev_pt = pt;
+        }
+      } else {
+        // unable to find path so just draw a fly-wire
+        int x, y;
+        source_node->xy(x, y);
+        odb::Point source_pt(x, y);
+        sink_node->xy(x, y);
+        odb::Point sink_pt(x, y);
+        painter.drawLine(source_pt, sink_pt);
+      }
+    }
+  }
+  painter.restorePenAndBrush();
+}
+
+void DbNetDescriptor::buildNodeMap(odb::dbWireGraph* graph, NodeMap& node_map) const
+{
+  for (auto itr = graph->begin_nodes(); itr != graph->end_nodes(); itr++) {
+    const auto* node = *itr;
+    NodeList connections;
+
+    int x, y;
+    node->xy(x, y);
+    const odb::Point node_pt(x, y);
+    const odb::dbTechLayer* layer = node->layer();
+
+    for (auto itr = graph->begin_edges(); itr != graph->end_edges(); itr++) {
+      const auto* edge = *itr;
+      const auto* source = edge->source();
+      const auto* target = edge->target();
+
+      if (source->layer() == layer || target->layer() == layer) {
+        int sx, sy;
+        source->xy(sx, sy);
+
+        int tx, ty;
+        target->xy(tx, ty);
+        if (sx > tx) {
+          std::swap(sx, tx);
+        }
+        if (sy > ty) {
+          std::swap(sy, ty);
+        }
+        const odb::Rect path_line(sx, sy, tx, ty);
+        if (path_line.intersects(node_pt)) {
+          connections.insert(source);
+          connections.insert(target);
+        }
+      }
+    }
+
+    node_map[node].insert(connections.begin(), connections.end());
+    for (const auto* cnode : connections) {
+      if (cnode == node) {
+        // don't insert the source node
+        continue;
+      }
+      node_map[cnode].insert(node);
+    }
+  }
+}
+
+void DbNetDescriptor::findPath(NodeMap& graph,
+                               const Node* source,
+                               const Node* sink,
+                               std::vector<odb::Point>& path) const
+{
+  // find path from source to sink using A*
+  // https://en.wikipedia.org/w/index.php?title=A*_search_algorithm&oldid=1050302256
```

Comment:
do you want &oldid=1050302256?

Reply:
I kept it since thats the permanent link, so if the page changes in the future, atleast the version I was referencing is that one.